zmk/build.py

```diff
@@ -181,12 +182,13 @@ def fix_key(key: str):
             return data
 
 
-def _to_yaml(item: BuildItem):
+# NOTE(elpekenin): cant hint return rtpe as BuildItem :(
+def _to_yaml(item: BuildItem) -> dict[str, Any]:
```

Comment:
Is type hinting private and inner functions necessary? I would think that automatic type inference would be fine for these, since they're not part of any public API.

Reply:
To clarify my reasoning on that, if you change the implementation of a private/inner function such that it returns a new type, then that should only break code within the same file, which you can easily identify and fix. Tooling will already figure everything out without them, so maintaining them is just busywork for little to no benefit.

However, if you change the return type of a public function, that is potentially a breaking change not just within the project (which tooling would identify) but to anything that decides to use zmk-cli as a dependency. Adding type hints to the return values prevents you from accidentally doing this. If such a change is intentional, needing to change the type hint is also a clue that a breaking change needs to be documented.

Reply:
There was no big thinking behind my changes, i just thought explicit typing was better than implicit. 

And sure, some tools will be smart enough to infer, but they may miss extra info we may add ourselves, and others may not try to infer anything.

I am not too sold on either alternative, so i can undo typing of private functions if you prefer that way